activerecord/lib/active_record/connection_adapters/role.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module ConnectionAdapters
+    class Role # :nodoc:
+      include Mutex_m
+
+      attr_reader :db_config, :connection_specification_name
+      attr_accessor :schema_cache
+
+      INSTANCES = ObjectSpace::WeakMap.new
+      private_constant :INSTANCES
+
+      class << self
+        def discard_pools!
+          INSTANCES.each_key(&:discard_pool!)
+        end
+      end
+
+      def initialize(connection_specification_name, db_config)
+        super()
+        @connection_specification_name = connection_specification_name
```

Comment:
I guess this is ok for now, but I'd really like to get rid of `connection_specification_name`.

IMHO it's backwards. It's not the `Role / DbConfig` that should have access to the root model identifier (`AnimalsBase`), but the root model that should know what `Role / DbConfig` it connects to (`animals_replica`)

Reply:
Definitely want to get rid of this but for now this seemed like a good compromise. Continue using it but name it something obvious that screams "get me out of here" 😅